src/execution/operator/persistent/physical_copy_to_file.cpp

```diff
@@ -57,13 +74,69 @@ SinkResultType PhysicalCopyToFile::Sink(ExecutionContext &context, GlobalSinkSta
 	return SinkResultType::NEED_MORE_INPUT;
 }
 
+static void CreateDir(ClientContext &context, const string &dir_path) {
+	auto &fs = FileSystem::GetFileSystem(context);
+	if (!fs.DirectoryExists(dir_path)) {
+		fs.CreateDirectory(dir_path);
+	}
+}
+
+static string CreateDirRecursive(ClientContext &context, const vector<idx_t> &cols, const vector<string> &names,
+                                 const vector<Value> &values, string path) {
+	CreateDir(context, path);
+
+	for (idx_t i = 0; i < cols.size(); i++) {
+		auto partition_col_name = names[cols[i]];
+		auto partition_value = values[i];
+		string p_dir = partition_col_name + "=" + partition_value.ToString();
+		path += "/" + p_dir;
```

Comment:
Should this not get the path separator from the file system, or use `fs.JoinPath`?